logstash-core/src/main/java/org/logstash/plugins/PluginFactoryExt.java

```diff
@@ -351,6 +352,10 @@ private IRubyObject plugin(final ThreadContext context, final PluginLookup.Plugi
         }
     }
 
+    private static Object unwrapIfNecessary(Object o) {
```

Comment:
Is the consequence of effectively overloading `PluginSpecification#klass`?

Previously, it always held a Java class because we initialised it with the result of `Object#javaClass`, but now it can hold a JRuby proxy object that holds a reference to a java class?

If possible, I would prefer that we resolve this discrepancy in either `PluginSpecification` or in the code that instantiates the `PluginSpecification`, instead of letting that complexity leak through to consumers of `PluginSpecification` as it does here.

Could we modi

Reply:
Yep, I got down into the weeds of Java classloaders and left that a little sloppy. The resolution was easy -- provide an instance of `Java::JavaClass` in registry.rb the same as it would if the default classloader were being used.